### PR TITLE
Tests for oe-gdb.

### PR DESCRIPTION
**Pull request by anakrish**
1. Create oe-gdb in current build's output folder.
   This allows running the current builds oe-gdb to debug
   enclaves in the current build. Previously, you'd have to
   do an install from the current build to get a working
   oe-gdb.
2. First set of tests for oe-gdb.
   Lock down:
     1. Setting multiple breakpoints (pending breakpoints).
     2. Read and modify variables.
     3. Call enclave functions from oe-gdb.
   Tests enabled in both Debug and RelWithDbgInfo builds.

### CODE REVIEW

**Comment by anakrish** [comment]
bors try

**Comment by anakrish** [comment]
bors try

**Comment by anakrish** [comment]
bors try

**Comment by anitagov** [comment]
Great job Anand on getting the first set of tests ready for the debugger.

File: tests/debugger/oe-gdb/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+add_subdirectory(host)
+add_subdirectory(enc)
+
+add_test(
```
Comment (radhikaj) [inline tests/debugger/oe-gdb/CMakeLists.txt]: Maybe skip this test in simulation mode?

Comment (anakrish) [reply tests/debugger/oe-gdb/CMakeLists.txt]: Good point. Done.
I will also later work on supporting oe-gdb in simulation mode.

**Comment by anitagov** [comment]
@anakrish - I like the first change as well since it is frustrating that we have to do a make install only to get the debugger to work.

**Pull request merged**